src/Microsoft.DotNet.Build.Tasks/Targets/packageresolve.targets

```diff
@@ -2,31 +2,39 @@
 <Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
 
   <UsingTask TaskName="ResolveNuGetPackages" AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"/>
+  <UsingTask TaskName="ResolveNuGetPackageAssets" AssemblyFile="$(ToolsDir)Microsoft.DotNet.Build.Tasks.dll"/>
 
   <PropertyGroup>
     <ProjectPackagesConfigFile Condition="'$(ProjectPackagesConfigFile)'=='' and Exists('$(MSBuildProjectDirectory)/packages.config')">$(MSBuildProjectDirectory)/packages.config</ProjectPackagesConfigFile>
+    <ProjectJson Condition="'$(ProjectJson)'=='' and Exists('$(MSBuildProjectDirectory)/project.json')">$(MSBuildProjectDirectory)/project.json</ProjectJson>
+    <ProjectLockJson>$(MSBuildProjectDirectory)/project.lock.json</ProjectLockJson>
```

Comment:
It's actually created by dnu.cmd restore next to the project.  The design of their system is to always put it next to the project.json.  The idea is that it is a cache of the asset applicability and can be checked in.  A scenario I'd like to use is switch to * versioned dependencies, resolve on dev box, generate .lock.json, check in.  Occasionally delete all .lock.jsons and regenerate to pick up new builds.

The property itself is no longer used.  It's left behind from a previous version of this change where I deleted the lock file after it was created (since it's currently not working and using it causes issues, as described in the change descriptions), but instead I switched to just exposing an argument to tell the package resolve task to ignore the file.

I do copy the project.json to intermediate and restore from there for the tests, that's something we could do across all of our projects if we wanted, but I think I'd like to actually check in and consume the lock file in the future.

For now I can remove the property.